nservicebus/operations/async-analyzer.md

```diff
@@ -0,0 +1,53 @@
+---
+title: Async Analyzer
+summary: How to use the async analyzer to avoid missing awaits
+component: Core
+versions: '[6,]'
+---
+
+Asynchronous APIs bring great performance improvements. However, they introduce new risks for bugs due to missed `await` statements. NServiceBus comes with a built in Roslyn analyzer to detect those missing awaits.
+
+## Functionality
+
+The analyzer will point out missing `await` statements when using asynchronous methods on the following classes and interfaces:
+* `IMessageHandlerContext`
+* `IMessageSession`
+* `Saga`
+* `IEndpointInstance`
+* `Endpoint`
+
+## Limitations
+
+This analyzer does only inspect the APIs contained on the classes and interfaces as described in the previous chapter. Other asynchronous operations on NServiceBus APIs are not analyzed.
+
+The analyzer only checks for ignored tasks returned from the scanned methods. If the task is assigned to a variable or passed to another method, the analyzer will not analyze this task's usage any further.
+
+Cases which are already handled by existing compiler warnings (e.g. [CS4014](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/cs4014)) are ignored by the NServiceBus analyzer.
```

Comment:
Should we mention something about enabling treat warnings as errors in this case? So many shops do not even know about this setting and will still be dropping tasks.

Reply:
we've removed that part of the code anyway. We will now also raise an error in that case. Might still be a good idea to recommend using the treat warnings as errors flag in addition.